src/units.rs

```diff
@@ -605,6 +614,20 @@ pub fn divide(left: Number, right: Number) -> Result<Number, String> {
     let hours = convert(right, Hour)?;
     let kph = Number::new(kilometers.value / hours.value, KilometersPerHour);
     Ok(convert(kph, final_unit)?)
+  } else if lcat == Length && rcat == Speed {
+    // 12 km / 100 kph
+    let final_unit = match (left.unit, right.unit) {
+      (Kilometer, KilometersPerHour) => Hour,
+      (Meter, MetersPerSecond) => Second,
+      (Mile, MilesPerHour) => Hour,
+      (Foot, FeetPerSecond) => Second,
+      (NauticalMile, Knot) => Hour,
+      _ => Hour,
+    };
+    let kilometers = convert(left, Kilometer)?;
+    let kilometers_per_hour = convert(right, KilometersPerHour)?;
+    let hour = Number::new(kilometers.value / kilometers_per_hour.value, Hour);
+    Ok(to_ideal_unit(convert(hour, final_unit)?))
```

Comment:
If we want to always get `Hour` when we divide `km/h`, I'd suggest making sure `to_ideal_unit()` doesn't run when we divide `km/h`. You could make that happen by having `final_unit` be an `Option`

Reply:
I don't think it's strictly necessary to return `Hour` when we divide by `KilometerPerHour`. Google doesn't do that when you query it with `20 km divided by 20 kph` for example.

However, I think it's important to be conservative in what transformations are applied, at least at first. That's why I've checked specifically for seconds, minute and hours.

Reply:
I agree that we don't need to return `Hour` when dividing by `kph`.

I don't think there's any advantage to making `to_ideal_unit()` conservative, because it's job isn't really to transform a unit, but rather decide a unit. For example, when we multiply `Length * Length`, we get the answer in `mm2`, then we pass that to `to_ideal_unit()`.

Reply:
Probably should rename `to_ideal_unit` to something like `decide_unit` so it's less confusing